infra/terraform/global/assets/prune_ebs_snapshots/snapshot/snapshot.go

```diff
@@ -0,0 +1,74 @@
+package snapshot
+
+import (
+	"../caller"
+	"fmt"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"log"
+	"time"
+)
+
+func GetSnapShots(connection *ec2.EC2, key string, val string) (output *ec2.DescribeSnapshotsOutput, err error) {
+
+	var snapShotInput *ec2.DescribeSnapshotsInput
+
+	// Get the owner ID a.k.a the aws account number
+	callerResult, callerErr := caller.GetCaller()
+	if callerErr != nil {
+		log.Panic(callerErr)
+	}
+	ownerID := *callerResult.Account
+
+	if len(key) != 0 && len(val) != 0 {
```

Comment:
I can see both branches of the `if` have the filter by `owner-id`, you could have the common part outside the `if` and have a positive `if` to add the `key`/`value` if necessary (you can therefor remove the variable declaration on top of the function, e.g. (not tested ⚠️ !)

```go
func GetSnapShots(connection *ec2.EC2, key string, val string) (output *ec2.DescribeSnapshotsOutput, err error) {
	// Get the owner ID a.k.a the aws account number
	caller, err := caller.GetCaller()
	if err != nil {
		log.Panicf("Error while getting owner ID: ", err)
	}
	ownerID := *caller.Account

	filters := []*ec2.Filter{
		{
			Name: aws.String("owner-id"),
			Values: []*string{
				aws.String(ownerID),
			},
		},
	}

	if len(key) > 0 && len(value) > 0 {
		filters = append(filters, &ec2.Filter{
			Name: aws.String(fmt.Sprintf("tag:%s", key)),
			Values: []*string{
				aws.String(val),
			},
		})
	}

	snapShotInput := &ec2.DescribeSnapshotsInput{Filters: filters}

	return connection.DescribeSnapshots(snapShotInput)
}
```

Reply:
Sorry, it should be: `log.Panicf("Error while getting owner ID: %s", err)`